php-frontend/src/test/java/org/sonar/php/tree/impl/expression/ExpandableStringLiteralTreeTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010 SonarSource and Akram Ben Aissi
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.php.tree.impl.expression;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import org.junit.Test;
+import org.sonar.php.PHPTreeModelTest;
+import org.sonar.php.parser.PHPLexicalGrammar;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.expression.ExpandableStringCharactersTree;
+import org.sonar.plugins.php.api.tree.expression.ExpandableStringLiteralTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+
+public class ExpandableStringLiteralTreeTest extends PHPTreeModelTest {
+
+  @Test
+  public void simple_variable() throws Exception {
+    ExpandableStringLiteralTree tree = parse("\"simple var: $a\"", PHPLexicalGrammar.ENCAPS_STRING_LITERAL);
+
+    assertExpandableStringLiteral(tree, 1, 1);
+
+    assertFirstString(tree, "simple var: ");
+    assertFirstExpression(tree, "$a", Kind.VARIABLE_IDENTIFIER);
+  }
+
+  private void assertExpandableStringLiteral(ExpandableStringLiteralTree tree, int stringsSize, int expressionsSize) {
+    assertThat(tree.is(Kind.EXPANDABLE_STRING_LITERAL)).isTrue();
+
+    assertThat(tree.openDoubleQuoteToken().text()).isEqualTo("\"");
+    assertThat(tree.strings()).hasSize(stringsSize);
+    assertThat(tree.expressions()).hasSize(expressionsSize);
+    assertThat(tree.closeDoubleQuoteToken().text()).isEqualTo("\"");
+  }
+
+  private void assertFirstExpression(ExpandableStringLiteralTree tree, String s, Kind kind) {
+    ExpressionTree expr = tree.expressions().get(0);
+
+    assertThat(expr.is(kind));
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Complete the assertion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2970)